nodejs/eks/nodegroup.ts

```diff
@@ -829,6 +829,59 @@ function createNodeGroupInternal(
             }
         });
 
+    const amiInfo = pulumi.output(amiId).apply((id) =>
+        aws.ec2.getAmi(
+            {
+                owners: ["self", "amazon"],
+                filters: [
+                    {
+                        name: "image-id",
+                        values: [id],
+                    },
+                ],
+            },
+            { parent },
+        ),
+    );
+
+    const { rootBlockDevice, ebsBlockDevices } = pulumi
+        .all([os, amiInfo])
+        .apply(([os, amiInfo]) => {
+            if (os !== OperatingSystem.Bottlerocket) {
+                return {
+                    rootBlockDevice: {
+                        encrypted: args.nodeRootVolumeEncrypted ?? false,
+                        volumeSize: args.nodeRootVolumeSize ?? 20, // GiB
+                        volumeType: args.nodeRootVolumeType ?? "gp2",
```

Comment:
Probably out of scope for this PR, but it would probably be good to update the default to `gp3`. I _think_ that the migration is non-breaking, but we would have to test
https://aws.amazon.com/blogs/storage/migrate-your-amazon-ebs-volumes-from-gp2-to-gp3-and-save-up-to-20-on-costs/

Reply:
Yeah in theory I agree with updating this here, the issue is that this will update the node groups and cause them to get cycled.
For self managed node groups this is an operation that will cause downtime when not done carefully (cordon & drain the nodes before running the update).

But this actually got me thinking. We shouldn't set any defaults if the OS is set to AL2023/Bottlerocket

But